packages/kbn-typed-react-router-config/src/create_router.test.tsx

```diff
@@ -252,6 +274,28 @@ describe('createRouter', () => {
         },
       });
     });
+
+    it('matches deep routes', () => {
+      history.push('/services/opbeans-java/errors/foo?rangeFrom=now-15m&rangeTo=now');
+
+      const matchedRoutes = router.matchRoutes(
+        '/services/{serviceName}/errors/{groupId}',
```

Comment:
What does this first argument do? Intuitively I'd think we would just compare the current url (`history.location`) against the registered routes.

Reply:
it's the route path, and it's needed for the types. `location.search` is just a string/value, we can't infer anything from that.